src/java/org/apache/cassandra/tcm/RemoteProcessor.java

```diff
@@ -200,51 +203,27 @@ public static <REQ, RSP> RSP sendWithCallback(Verb verb, REQ request, CandidateI
 
     public static <REQ, RSP> void sendWithCallbackAsync(Promise<RSP> promise, Verb verb, REQ request, CandidateIterator candidates, Retry retryPolicy)
     {
-        class Request implements RequestCallbackWithFailure<RSP>
```

Comment:
@ifesdjeen if you want I can revert, I tried to make sure the subtle semantics did match (other than allocation of callback), but figured could have all 3 usage use 1 thing that has tests rather than all 3 locations doing custom logic